packages/swap/src/utils/ServiceError.ts

```diff
@@ -5,30 +5,43 @@ type IgnoredField =
   | 'prepareStackTrace'
   | 'stackTraceLimit';
 
+type ErrorCode = 'not-found' | 'bad-request' | 'internal-error' | 'rpc-error' | 'invalid-amount';
+
 export default class ServiceError extends Error {
-  static badRequest(message: string): ServiceError {
-    return new ServiceError(message, 400);
+  static badRequest(
+    code: ErrorCode = 'bad-request',
+    message = 'the request payload is not valid',
```

Comment:
I think the message will specified more often than the code
```suggestion
    message = 'the request payload is not valid',
    code: ErrorCode = 'bad-request',
```